sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -480,10 +480,11 @@ class DataFrame private[sql](
    *
    * @param right Right side of the join operation.
    * @param usingColumns Names of the columns to join on. This columns must exist on both sides.
+   * @param joinType One of: (default)`inner`, `outer`, `left_outer`, `right_outer`, `leftsemi`.
    * @group dfops
    * @since 1.4.0
    */
-  def join(right: DataFrame, usingColumns: Seq[String]): DataFrame = {
+  def join(right: DataFrame, usingColumns: Seq[String], joinType: String = "inner"): DataFrame = {
```

Comment:
We cannot use default parameter values in order to maintain compatibility with Java. You can add an extra method.